via/views/view_video.py

```diff
@@ -8,26 +9,44 @@
 from via.exceptions import BadURL
 from via.security import ViaSecurityPolicy
 from via.services import YouTubeService
+from via.services.youtube_api import CaptionTrack
+
+CAPTION_TRACK_PREFERENCES = (
+    # Plain English first
+    CaptionTrack(language_code="en"),
+    # Plain varieties of English
+    CaptionTrack(language_code=Any.string.matching("^en-.*")),
+    # Sub-categories of plain English
+    CaptionTrack(language_code="en", name=Any()),
+    # English varieties with names
+    CaptionTrack(language_code=Any.string.matching("^en-.*"), name=Any()),
+    # Any auto generated English
+    CaptionTrack(language_code=Any.string.matching("^en-.*"), name=Any(), kind=Any()),
+    # Any track which can be translated into English
+    CaptionTrack(
+        language_code=Any(), name=Any(), kind=Any(), translated_language_code="en"
+    ),
+)
 
 
 @view_config(renderer="via:templates/view_video.html.jinja2", route_name="youtube")
 @use_kwargs(
     {"url": fields.Url(required=True)}, location="query", unknown=marshmallow.INCLUDE
 )
-def youtube(request, url, **kwargs):
-    youtube_service = request.find_service(YouTubeService)
+def view_youtube_video(request, url, **kwargs):
+    youtube_service: YouTubeService = request.find_service(YouTubeService)
 
     if not youtube_service.enabled:
         raise HTTPUnauthorized()
 
     video_id = youtube_service.get_video_id(url)
-    video_url = youtube_service.canonical_video_url(video_id)
-    video_title = youtube_service.get_video_title(video_id)
-
     if not video_id:
         raise BadURL(f"Unsupported video URL: {url}", url=url)
 
-    _, client_config = Configuration.extract_from_params(kwargs)
+    video_url = youtube_service.canonical_video_url(video_id)
+    video_title = youtube_service.get_video_title(video_id)
```

Comment:
I moved these down, because we can early return if the video id isn't valid. Rather than doing lots of strange stuff.